### PR TITLE
Fix using RealmModel as a field

### LINKED ISSUE #2654: Field that implements RealmModel
Can I use fields that implements RealmModel? Thanks!

### PR DESCRIPTION
**Pull request by beeender**
* Modify and rename isRealmObject to isRealmModel.
* Modify annotation processor to allow RealmModel as a field.

Fix #2654

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ### Bug fixes
 
 * @PrimaryKey + @Required on String type primary key no longer throws when using copyToRealm or copyToRealmOrUpdate (#2653).
+* Objects implements RealmModel can be used as a field of RealmModel/RealmObject (#2654).
```
Comment (kneth) [inline CHANGELOG.md]: "Objects implementing RealmModel"

**Comment by kneth** [comment]
:+1:

**Comment by stk1m1** [comment]
:+1:

**Comment by stk1m1** [comment]
CI: retest this

**Comment by kneth** [comment]
Still :+1: Maybe I should do a search for `RealmObject`?

**Pull request merged**